modules/core/src/lib/attribute.js

```diff
@@ -206,18 +208,26 @@ export default class LayerAttribute extends Attribute {
           this.update({constant: false, buffer});
           state.needsRedraw = true;
         }
-      } else {
-        const ArrayType = glArrayFromType(this.type || GL.FLOAT);
-        if (!(buffer instanceof ArrayType)) {
-          throw new Error(`Attribute ${this.id} must be of type ${ArrayType.name}`);
-        }
+      } else if (this.value !== buffer) {
         if (state.auto && buffer.length <= numInstances * this.size) {
           throw new Error('Attribute prop array must match length and size');
         }
-        if (this.value !== buffer) {
+
+        const ArrayType = glArrayFromType(this.type || GL.FLOAT);
+        if (!(buffer instanceof ArrayType)) {
+          log.warn(
+            `Attribute ${this.id} is type ${buffer.constructor.name}, expecting type ${
```

Comment:
These are pretty extensive error messages, normally we don't do this due to e.g. bundle size concerns from the strings etc/

Maybe it is justified here though?

Also `constructor.name` gets clobbered during minification so this would generated random message in that case, better avoid.

Reply:
Since `buffer` is a typed array constructor.name should not be mangled